week7/answers.sql

```diff
@@ -1,3 +1,207 @@
 -- Create your tables, views, functions and procedures here!
 CREATE SCHEMA destruction;
 USE destruction;
+
+CREATE TABLE players (
+    player_id INT AUTO_INCREMENT PRIMARY KEY,
+    first_name VARCHAR(50),
+    last_name VARCHAR(50),
+    email VARCHAR(100)
+);
+
+CREATE TABLE characters (
+    character_id INT AUTO_INCREMENT PRIMARY KEY,
+    player_id INT,
+    name VARCHAR(100),
+    level INT,
+    FOREIGN KEY (player_id) REFERENCES players(player_id)
+);
+
+CREATE TABLE winners (
+    character_id INT AUTO_INCREMENT PRIMARY KEY,
+    name VARCHAR(100),
+    FOREIGN KEY (character_id) REFERENCES characters(character_id)
+);
+
+CREATE TABLE character_stats (
+    character_id INT AUTO_INCREMENT PRIMARY KEY,
+    health INT,
+    armor INT,
+    FOREIGN KEY (character_id) REFERENCES characters(character_id)
+);
+
+CREATE TABLE teams (
+    team_id INT AUTO_INCREMENT PRIMARY KEY,
+    name VARCHAR(100)
+);
+
+CREATE TABLE team_members (
+    team_member_id INT AUTO_INCREMENT PRIMARY KEY,
+    team_id INT,
+    character_id INT,
+    FOREIGN KEY (team_id) REFERENCES teams(team_id),
+    FOREIGN KEY (character_id) REFERENCES characters(character_id)
+);
+
+CREATE TABLE items (
+    item_id INT AUTO_INCREMENT PRIMARY KEY,
+    name VARCHAR(100),
+    armor INT,
+    damage INT
+);
+
+CREATE TABLE inventory (
+    inventory_id INT AUTO_INCREMENT PRIMARY KEY,
+    character_id INT,
+    item_id INT,
+    FOREIGN KEY (character_id) REFERENCES characters(character_id),
+    FOREIGN KEY (item_id) REFERENCES items(item_id)
+);
+
+CREATE TABLE equipped (
+    equipped_id INT AUTO_INCREMENT PRIMARY KEY,
+    character_id INT,
+    item_id INT,
+    FOREIGN KEY (character_id) REFERENCES characters(character_id),
+    FOREIGN KEY (item_id) REFERENCES items(item_id)
+);
+
+CREATE VIEW character_items AS
+SELECT DISTINCT inv.character_id, c.name AS character_name, i.name AS item_name, i.armor, i.damage
+FROM inventory inv
+JOIN characters c ON inv.character_id = c.character_id
+JOIN items i ON inv.item_id = i.item_id
+UNION
+SELECT DISTINCT eq.character_id, c.name AS character_name, i.name AS item_name, i.armor, i.damage
+FROM equipped eq
+JOIN characters c ON eq.character_id = c.character_id
+JOIN items i ON eq.item_id = i.item_id;
+
+CREATE VIEW team_items AS
+SELECT tm.team_id, t.name AS team_name, i.name AS item_name, i.armor, i.damage
+FROM team_members tm
+JOIN teams t ON tm.team_id = t.team_id
+JOIN characters c ON tm.character_id = c.character_id
+JOIN inventory inv ON c.character_id = inv.character_id
+JOIN items i ON inv.item_id = i.item_id
+UNION
+SELECT tm.team_id, t.name AS team_name, i.name AS item_name, i.armor, i.damage
+FROM team_members tm
+JOIN teams t ON tm.team_id = t.team_id
+JOIN characters c ON tm.character_id = c.character_id
+JOIN equipped eq ON c.character_id = eq.character_id
+JOIN items i ON eq.item_id = i.item_id;
+
+DELIMITER //
+CREATE FUNCTION armor_total(character_id INT) RETURNS INT
```

Comment:
As the automated GitHub tests mention, this FUNCTION isn't fully defined. It is missing syntax required for a FUNCTION.